src/characters/universal_bob.ts

```diff
@@ -0,0 +1,108 @@
+import {
+  EncryptedTreasureMap,
+  MessageKit,
+  PublicKey,
+  SecretKey,
+  Signer,
+} from '@nucypher/nucypher-core';
+
+import { Keyring } from '../keyring';
+import { PolicyMessageKit } from '../kits/message';
+import { RetrievalResult } from '../kits/retrieval';
+import { zip } from '../utils';
+
+import { Porter } from './porter';
+
+export class tDecDecrypter {
+  private readonly porter: Porter;
+  private readonly keyring: Keyring;
+  private readonly policyEncryptingKey: PublicKey;
+  private readonly encryptedTreasureMap: EncryptedTreasureMap;
+  private readonly publisherVerifyingKey: PublicKey;
+  private readonly verifyingKey: Keyring;
+
+  constructor(
+    porterUri: string,
+    policyEncryptingKey: PublicKey,
+    encryptedTreasureMap: EncryptedTreasureMap,
+    publisherVerifyingKey: PublicKey,
+    secretKey: SecretKey,
+    verifyingKey: SecretKey
+  ) {
```

Comment:
Just a thought:
TypeScript has a shorthand for creating and assigning a class instance property from a constructor parameter.
```suggestion
  constructor(
    private readonly porter: Porter;
    private readonly keyring: Keyring;
    private readonly policyEncryptingKey: PublicKey;
    private readonly encryptedTreasureMap: EncryptedTreasureMap;
    private readonly publisherVerifyingKey: PublicKey;
    private readonly verifyingKey: Keyring;
  ) {
```